src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -324,16 +357,16 @@ fn build_header_links(html: String, filename: &str) -> String {
             }
             let id = id.chars()
                 .filter_map(|c| if c.is_alphanumeric() || c == '-' || c == '_' {
-                                if c.is_ascii() {
-                                    Some(c.to_ascii_lowercase())
-                                } else {
-                                    Some(c)
-                                }
-                            } else if c.is_whitespace() && c.is_ascii() {
-                                Some('-')
-                            } else {
-                                None
-                            })
+                    if c.is_ascii() {
```

Comment:
I just had a shot at fixing this up. The issue I'm having is that there aren't any tests to go off of and it's difficult to understand what the underlying algorithm is.

I *think* it's trying to:

- generate an id for each header tag (h1, h2, ...) from its content by
  - removing any inner tags
  - ascii lowercasing
  - replacing spaces with dashes
  - appending an auto-incremented counter to make sure the id is unique
- then wrap the entire thing in an anchor tag so regex can replace the old header with the new one

Can you or @azerupi let me know if I'm on the right track?